net/devif/ipv6_input.c

```diff
@@ -609,6 +609,12 @@ int ipv6_input(FAR struct net_driver_s *dev)
   FAR uint8_t *buf;
   int ret;
 
+  /* Store reception timestamp if enabled and not provided by hardware. */
+
+#if defined(CONFIG_NET_TIMESTAMP) && !defined(CONFIG_ARCH_HAVE_NETDEV_TIMESTAMP)
+  clock_gettime(CLOCK_REALTIME, &dev->d_rxtime);
```

Comment:
let's create a new patch to update socketcan to allow the net can driver return timestamp too

Reply:
SocketCAN does something similar [here](https://github.com/apache/nuttx/blob/master/net/can/can_callback.c#L136-L164) but I'm unfamiliar with CAN and have no way to test it, so I cannot help in updating it.